app/components/primer/dropdown.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+module Primer
+  # `Dropdown` is a lightweight context menu for housing navigation and actions.
+  # They're great for instances where you don't need the full power (and code) of the select menu.
+  class Dropdown < Primer::Component
+    # Required trigger for the dropdown. Has the same arguments as <%= link_to_component(Primer::ButtonComponent) %>,
+    # but it is locked as a `summary` tag.
+    renders_one :summary, lambda { |**system_arguments, &block|
+      @summary_arguments = system_arguments
+      @summary_arguments[:button] = true
+
+      view_context.capture { block&.call }
+    }
+
+    # Required context menu for the dropdown
+    #
+    # @param as [Symbol] When `as` is `:list`, wraps the menu in a `<ul>` with a `<li>` for each item.
+    # @param direction [Symbol] <%= one_of(Primer::Dropdown::Menu::DIRECTION_OPTIONS) %>
+    # @param scheme [Symbol] Pass `:dark` for dark mode theming
+    # @param header [String] Optional string to display as the header
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    renders_one :menu, "Primer::Dropdown::Menu"
+
+    # @example Default
+    #   <div>
+    #     <%= render(Primer::Dropdown.new) do |c| %>
+    #       <% c.summary do %>
```

Comment:
In what cases do you think we should use `with_content` in our examples? I think it would be nice to use here.

Reply:
The problem of using `with_content` here is how slots work :/
Since we aren't using any arguments in this slot, the method acts as a getter, so it returns `nil`

Reply:
Ah yes, we are tracking a bug for that.